types/datum.go

```diff
@@ -1009,6 +1009,19 @@ func (d *Datum) convertToMysqlDuration(sc *stmtctx.StatementContext, target *Fie
 		if err != nil {
 			return ret, errors.Trace(err)
 		}
+	case KindInt64, KindFloat32, KindFloat64, KindMysqlDecimal:
+		// TODO: We need a ParseDurationFromNum to avoid the cost of converting a num to string.
```

Comment:
maybe we need create a lira issue rather than adding a "TODO". People are all joking about "TODO" is just another expression of "never do". lol

Reply:
Ok, I'll create a jira issue after this PR is merged.